syntax = "proto3";

package chat;

service ChatService {
  rpc addMessageEvent(MessageEvent) returns (EmptyResponse) {}
  rpc getMessageEvent(GetMessageRequest) returns (stream MessageEvent) {}
  rpc deleteMessageEvent(MessageEvent) returns (EmptyResponse) {}
  rpc notifyNewMessage(NotifyEvent) returns (EmptyResponse) {}
}

message MessageEvent {
  int64 chat_id = 2;
  int64 user_1 = 3;
  int64 user_2 = 4;
  string message_text = 5;
}

message GetMessageRequest {
  int64 chat_id = 1;
}

message NotifyEvent {
    int64 not_id = 1;
    int64 message_id = 2;
    int64 chat_id = 3;
}

message EmptyResponse {}

